Unit-01/02-flask-routing/app.py

```diff
@@ -0,0 +1,34 @@
+from flask import Flask
+app = Flask(__name__)
+
+@app.route("/add/<int:add_num1>/<int:add_num2>")
+def add(add_num1, add_num2):
+	return "The sum is {}".format(add_num1 + add_num2)
+
+@app.route("/subtract/<int:subtract_num1>/<int:subtract_num2>")
+def subtract(subtract_num1, subtract_num2):
+		return "The difference is {}".format(subtract_num1 - subtract_num2) 
+
+@app.route("/multiply/<int:mult_num1>/<int:mult_num2>")
+def multiply(mult_num1, mult_num2):
+	return "The product is {}".format(mult_num1 * mult_num2)
+
+@app.route("/divide/<int:divide_num1>/<int:divide_num2>")
+def divide(divide_num1, divide_num2):
+	return "The quotient is {}".format(divide_num1 / divide_num2)
+
+@app.route("/math/<calculate>/<int:num1>/<int:num2>")
+def math(calculate, num1, num2):
+	if calculate == 'add':
+		return "The sum is {}".format(num1 + num2)
+	elif calculate == 'subtract': 	
+		return "The difference is {}".format(num1 - num2)
+	elif calculate == 'multiply': 	
```

Comment:
This is good; However, another way to do this is using a dictionary, just like you can in JavaScript using objects.

```python
def add(a, b):
    return a + b
def subtract(a, b):
    return a - b
def multiply(a, b):
    return a * b
def divide(a / b):
    return a / b

operations = {
    "add": add,
    "subtract": subtract,
    "multiply": multiply,
    "divide": divide
}

# @route handler
def math(operation, num1, num2):
    if operation in operations:
        return f"{operations[operation](num1, num2)}"
```

This is a more declarative style, where you can declare your functions and data structures at the time and simplify the business logic part (i.e. the routing, in this case)